sql-plugin/pom.xml

```diff
@@ -49,45 +49,457 @@
+                        <artifactId>hadoop-mapreduce-client-core</artifactId>
+                    </exclusion>
+                    <exclusion>
+                        <groupId>org.apache.orc</groupId>
+                        <artifactId>orc-core</artifactId>
+                    </exclusion>
+                    <exclusion>
+                        <groupId>org.apache.hive</groupId>
+                        <artifactId>hive-storage-api</artifactId>
+                    </exclusion>
+                    <exclusion>
+                        <groupId>com.esotericsoftware</groupId>
+                        <artifactId>kryo-shaded</artifactId>
+                    </exclusion>
+                    <exclusion>
+                        <groupId>com.google.code.findbugs</groupId>
+                        <artifactId>jsr305</artifactId>
+                    </exclusion>
+                </exclusions>
+            </dependency>
+            <dependency>
+                <groupId>org.apache.orc</groupId>
+                <artifactId>orc-shims</artifactId>
+                <exclusions>
+                    <exclusion>
+                        <groupId>org.apache.hadoop</groupId>
+                        <artifactId>hadoop-common</artifactId>
+                    </exclusion>
+                    <exclusion>
+                        <groupId>org.apache.hadoop</groupId>
+                        <artifactId>hadoop-hdfs</artifactId>
+                    </exclusion>
+                </exclusions>
+            </dependency>
+            <dependency>
+                <groupId>org.apache.hive</groupId>
+                <artifactId>hive-storage-api</artifactId>
+                <exclusions>
+                    <exclusion>
+                        <groupId>org.slf4j</groupId>
+                        <artifactId>slf4j-api</artifactId>
+                    </exclusion>
+                </exclusions>
+            </dependency>
+        </dependencies>
+    </dependencyManagement>
+
     <profiles>
+        <profile>
+            <id>release301</id>
+            <activation>
+                <property>
+                    <name>buildver</name>
+                    <value>301</value>
+                </property>
+            </activation>
+            <dependencies>
+                <dependency>
+                    <groupId>org.apache.orc</groupId>
+                    <artifactId>orc-core</artifactId>
+                    <version>1.5.10</version>
+                </dependency>
+                <dependency>
+                    <groupId>org.apache.orc</groupId>
+                    <artifactId>orc-mapreduce</artifactId>
+                    <version>1.5.10</version>
+                </dependency>
+                <dependency>
+                    <groupId>org.apache.hive</groupId>
+                    <artifactId>hive-storage-api</artifactId>
+                    <version>2.7.1</version>
+                </dependency>
```

Comment:
If the intent is to use the same version of ORC and Hive that Spark is using, why are we pulling in these dependencies explicitly rather than getting them transitively from the Spark artifacts?  The risk of putting them explicitly here is that they change versions for Spark snapshot artifacts and we fail to notice.

Reply:
Still waiting for an answer to this question.

Reply:
The scope of the transitive dependencies of Spark are provided, the maven shade plugin only shade compile jars, so should  explicitly specify orc jar as compile scope, and then should add the version.
And sometimes the transitive jars are not the same as the Spark runtime(spark-home/jars).

Your are right, for the Spark snapshot, the transitive jar will change without notice. Seems no good idea but mannully update the version.

Reply:
For the Spark snapshot, seems it's better to add a daily crontab job to check the version.

Reply:
An alternative would be to have the build run a script to compute the version by examining the Spark jar dependencies to find the ORC version.  It could then fail the build (or emit a loud warning) if it finds the computed ORC version doesn't match the manually specified version.  Even better would be to use the computed ORC version directly as the plugin's ORC dependency version if that's possible.